Assets/Scripts/SaveLoad/EntityModels/EnemyData.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using UnityEngine;
+
+public class EnemyData : EntityData
+{
+	public int Health { get => GetComponent<Health>().Current; }
+	public string AnimationName { get => GetComponentInChildren<Animator>().GetCurrentAnimatorClipInfo(0)[0].clip.name; }
+
+	public override EntitySavedData GetData()
+	{
+		var data = base.GetData();
+
+		return new EnemySavedData 
+		{
+			Id = data.Id,
+			Position = data.Position,
+			health = Health,
+			animationName = AnimationName,
```

Comment:
The difference in capitalization of the attributes between EntitySavedData and EnemySavedData is slightly confusing to me, is it supposed to be this way?

Reply:
My bad. Fixed. Sorry it;s because build in json converter doesn't support properties and is very restricted and I switched to Newtonsoft midway. Thanks for detailed review.